plugin/eleline.vim

```diff
@@ -191,6 +193,20 @@ function! s:hi_statusline()
   call s:hi('User9'      , 251 , s:bg+5 )
 endfunction
 
+function! InsertStatuslineColor(mode)
+  if a:mode == 'i'
+    call s:hi('User1' , 239 ,  179 )
+  elseif a:mode == 'r'
+    call s:hi('User1' , 239 ,  196 )
+  else
+    call s:hi('User1' , 7   , s:bg+8 )
+  endif
+endfunction
+
+au InsertEnter * call InsertStatuslineColor(v:insertmode)
```

Comment:
Wrap these `au` in `augroup eleline` below, see https://www.reddit.com/r/vim/wiki/vimrctips#wiki_wrap_your_autocmds_in_proper_augroups.